### PR TITLE
Yarn

### PR DESCRIPTION
**Pull request by jtormey**
This weekend, a dependency (possibly of a dependency) updated and made it so that `npm test` would hang after a few seconds... still not sure which one it was, not an easy thing to track down. This kind of issue is pretty annoying, so I'd like to switch us over to Yarn. Yarn fixes this issue by making sure each install is the same as the last one (provided no packages were manually updated).

Cool features:
- Deterministic installs, the yarn.lock file guarantees that every one will have the exact same node_modules
- Sooo much faster (I'm seeing 12-14s installs)
- Secure, yarn verifies package checksums while installing to make sure nothing slips by

I'm pretty excited about this, more about Yarn [here](https://yarnpkg.com/) 🚀

### CODE REVIEW

**Comment by mpfluger** [comment]
Very excited about this. Please add the install steps for yarn to the README.

File: README.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ JavaScript Model for Blockchain.info wallet.
 
 ## Build
 
+Install Yarn: `npm install -g yarn`
```
Comment (mpfluger) [inline README.md]: This is not the preferred way, see: https://yarnpkg.com/en/docs/install#alternatives-tab. You could either just link to that or change it to `brew install yarn` (only for Mac).

Comment (jtormey) [inline README.md]: Didn't see that earlier, thanks!

**Comment by Sjors** [comment]
I rebased this branch on top of v3.31-release

**Comment by Sjors** [comment]
The Makefile should also use yarn.

**Pull request merged**